src/pegins.cpp

```diff
@@ -533,3 +533,46 @@ CScriptWitness CreatePeginWitness(const CAmount& value, const CAsset& asset, con
 {
     return CreatePeginWitnessInner(value, asset, genesis_hash, claim_script, tx_ref, merkle_block);
 }
+
+bool DecomposePeginWitness(const CScriptWitness& witness, CAmount& value, CAsset& asset, uint256& genesis_hash, CScript& claim_script, boost::variant<boost::blank, Sidechain::Bitcoin::CTransactionRef, CTransactionRef>& tx, boost::variant<boost::blank, Sidechain::Bitcoin::CMerkleBlock, CMerkleBlock>& merkle_block)
+{
+    if (witness.stack.size() < 5) return false;
+
+    const auto& stack = witness.stack;
+
+    CDataStream stream(witness.stack[0], SER_NETWORK, PROTOCOL_VERSION);
```

Comment:
Should this just be `stack` (and elsewhere)?

Reply:
Yes, done.